admin/jobs/__init__.py

```diff
@@ -0,0 +1,132 @@
+from datetime import datetime
+from dateutil.relativedelta import relativedelta
+from flask import Blueprint, request
+from sqlalchemy.sql import extract
+from grc.utils.decorators import JobTokenRequired
+from grc.models import db, Application, ApplicationStatus, SecurityCode
+from grc.utils.application_files import ApplicationFiles
+from grc.external_services.gov_uk_notify import GovUkNotify
+
+jobs = Blueprint('jobs', __name__)
+
+
+@jobs.route('/jobs', methods=['GET'])
+def index():
+    return ('', 200)
+
+
+@jobs.route('/jobs/create-files', methods=['GET'])
+@JobTokenRequired
+# https://stackoverflow.com/questions/70791034/how-to-ssh-into-a-container-in-which-a-task-is-running
+def create_files():
+    applications = Application.query.filter_by(
+        status=ApplicationStatus.SUBMITTED,
+        filesCreated=False
+    )
+
+    for application in applications:
+        print('Creating attachments zipfile for application %s' % application.reference_number, flush=True)
+        ApplicationFiles().create_or_download_attachments(
+            application.reference_number,
+            application.data(),
+            download=False
+        )
+        print('Creating pdf for application %s' % application.reference_number, flush=True)
+        ApplicationFiles().create_or_download_pdf(
+            application.reference_number,
+            application.data(),
+            download=False
+        )
+
+        application.filesCreated = True
+        db.session.commit()
+
+    return ('', 200)
```

Comment:
Is this job needed?

I like the way that, in this PR, you create the PDF and ZIP files when the user clicks "Submit".
That's a really nice time to created these files 🙂 

Then, there's the backup option of generating the PDF and ZIP on-demand when an admin user downloads the files.
Do we also need another backup on generating the files nightly?

Assuming the files will almost always successfully be generate when the user clicks "Submit", then this nightly job will normally not do anything.
We've introduced a new column in the database to track this - we could avoid this if we decide we don't need the nightly job

Reply:
Strictly speaking it's not required, these files should generate in the background as soon as someone submits their application, and again when an admin requests the files.
The reason it's there is as a contingency against large files that could fail to generate in the normal way